src/main/java/org/openrewrite/jenkins/github/AddTeamToCodeowners.java

```diff
@@ -0,0 +1,165 @@
+        PlainTextParser parser = new PlainTextParser();
+        return parser.parse("* " + acc.teamName())
+                .map(brandNewFile -> (PlainText) brandNewFile.withSourcePath(Paths.get(FILE_PATH)))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(Scanned acc) {
+        return new PlainTextVisitor<ExecutionContext>() {
+            @Override
+            public PlainText visitText(PlainText text, ExecutionContext executionContext) {
+                if (acc.presentIn(text.getText())) {
+                    return text;
+                }
+                List<String> lines = new LinkedList<>();
+                List<String> after = new LinkedList<>();
+                Scanner scanner = new Scanner(text.getText());
+                int atPos = 0;
+                boolean lastComment = true;
+                while (scanner.hasNextLine()) {
+                    String line = scanner.nextLine();
+                    if (atPos == 0 && line.contains("@")) {
+                        atPos = line.indexOf("@");
+                    }
+                    if (lastComment && line.startsWith("#")) {
+                        lines.add(line);
+                    } else {
+                        lastComment = false;
+                        after.add(line);
+                    }
+                }
+                int spaces = Math.max(1, atPos - 1);
+                lines.add("*" + StringUtils.repeat(" ", spaces) + acc.teamName());
+                lines.addAll(after);
+                return text.withText(String.join("\n", lines));
+            }
+        };
+    }
+
+    @Data
+    public static class Scanned {
+        private final TeamNameGenerator<TeamNameInput> generator;
+        String artifactId;
+        boolean foundFile;
+
+        public Scanned(TeamNameGenerator<TeamNameInput> generator) {
+            this.generator = generator;
+        }
+
+        boolean presentIn(String text) {
+            Pattern p = Pattern.compile("^\\*\\s+" + teamName() + "$");
+            try (Scanner s = new Scanner(text)) {
+                while (s.hasNextLine()) {
+                    String line = s.nextLine();
+                    Matcher matcher = p.matcher(line);
+                    if (matcher.matches()) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        }
+
+        String teamName() {
+            return generator.generate(new TeamNameInput(artifactId));
+        }
+    }
+
+    private static class ArtifactIdExtractor extends MavenIsoVisitor<ExecutionContext> {
+        private String artifactId = "";
+
+        @Override
+        public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext executionContext) {
+            Xml.Tag t = super.visitTag(tag, executionContext);
+            if ("artifactId".equals(t.getName()) && !isManagedDependencyTag() && !isDependencyTag()) {
```

Comment:
Out of a slight worry that we might match for instance `//project/build/plugins/plugin/dependencies/dependency/artifactId` I figured it might be best to use an XPathMatcher here.
```suggestion
        private static final XPathMatcher PROJECT_ARTIFACTID_MATCHER = new XPathMatcher("/project/artifactId");

        @Override
        public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext executionContext) {
            Xml.Tag t = super.visitTag(tag, executionContext);
            if (PROJECT_ARTIFACTID_MATCHER.matches(getCursor())) {
```